.github/workflows/main.yml

```diff
@@ -0,0 +1,25 @@
+name: Build and Test
+on:
+  push:
+    branches:
+      - "*"
```

Comment:
CI should trigger only on creation of PR towards the main branch. Please change this if the rule is not being followed by the configuration.

Reply:
fix this to run ci only on main branch